h2/src/main/org/h2/mvstore/MVStore.java

```diff
@@ -1781,67 +1786,67 @@ private void compactMoveChunks(Iterable<Chunk> move) {
             // now re-use the empty space
             reuseSpace = true;
             for (Chunk c : move) {
-                moveChunk(c, false);
+                moveChunk(c, 0);
```

Comment:
OK: the first step of the moving has over and been persisted. So we can reuse the free space in the first step.